src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHouseDownload/DownloadStatusStorage.cs

```diff
@@ -101,21 +100,39 @@ public async Task CreateCompaniesHouseFileDownloadLogAsync(string partitionKey)
                         DownloadStatus = null
                     });
                 }
-
-                var batchTransactions = new List<TableTransactionAction>();
```

Comment:
This section only needs to run `if (entities.Count > 0)` - otherwise it will fail to run a transaction on an empty batch.